src/Servers/Kestrel/Core/src/KestrelConfigurationLoader.cs

```diff
@@ -313,6 +319,11 @@ public void Load()
                     configureEndpoint(endpointConfig);
                 }
 
+                if (https && endpoint.SslProtocols.HasValue)
```

Comment:
Move this above configureEndpoint so the config value is visible in the callback.

Reply:
Do you propose to move it or have it in both places (as was proposed in the other discussion)?
If I move it, the selection of SslProtocols would be inconsistent:

1.     Code default
2.     Configuration from EndpointDefaults section
3.     Configuration from named endpoint section
4.     Code configuration for named endpoint

Reply:
We should only have to set this once. I edited my original comment to indicate what we now want:

I would add the following right before ApplyHttpsDefaults:

```C#
httpsOptions.SslProtocols = ConfigurationReader.EndpointDefaults.SslProtocols ?? SslProtocols.None;
```

Then, right after ApplyHttpsDefault, I would add the following:

```C#
if (endpoint.SslProtocols.HasValue)
{
    httpsOptions.SslProtocols = endpoint.SslProtocols;
}
```

It's important that this second part happens before finding `matchingBoundEndpoints` or calling the `configureEndpoint` callbacks for named endpoints.

Reply:
Thank you, I was working with the wrong assumption that config should be applied after the code and got confused by the other discussion. Updated